framework/utils/utils.py

```diff
@@ -1109,3 +1109,40 @@ def getAllSubclasses(cls):
     @ Out, getAllSubclasses, list of class objects for each subclass of cls.
   """
   return cls.__subclasses__() + [g for s in cls.__subclasses__() for g in getAllSubclasses(s)]
+
+def which(cmd):
+  """
+    Emulate the which method in shutil
```

Comment:
Can you expand the description of this function?

Reply:
done